### PR TITLE
Update request url and handlers to support featured products

### PR DESCRIPTION
**Pull request by lhoffbeck**
### PR Summary: 

Update product info class to support featured product urls

### Why are these changes introduced?

Featured products are not currently supported in combined listings early access, due to pending backend work to return the combined listing product associated with the selected option values instead of simply returning the product from the section settings.

Once the backend is supported, Dawn should be in a position to work as expected. This change fixes a bug where the product info class requests the incorrect url for a featured product section.

**Behavior for main-product section requests**
- Request the full `/<product-handle>` page if switching between option values across multiple products (i.e. combined listing parent product options). On response, replace the content of `<main>`.
- Request the `/<product-handle>?section_id=<section_id>__main` if switching between option values within a single product. On response, replace the content of the main-product section.

**Behavior for featured-product section requests**
- Regardless of the page the featured product is embedded in, request the product page for the featured product with the featured product's dynamic sectionID and selected option values. On response, replace the content of the featured-product section.

### What approach did you take?

Refactored some of `ProductInfo#handleOptionValueChange` to account for featured products

### Visual impact on existing themes

None

### Testing steps/scenarios

> ⚠️  The test shop has a number of themes. Please test by previewing theme pr_3497.

Because featured products are not yet supported in the section rendering API, testing involves ensuring that the correct network requests are fired. We should test the following scenarios to cover all use cases:

**Main product - PDP**
- [ ] On [this product](https://combined-listings-test.myshopify.com/products/chinos-blue?variant=47651696836910), click on `color: orange`. The page should request `/chinos-orange?option_values=<id_1>,<id_2>,<id_3>`
- [ ] On [this product](https://combined-listings-test.myshopify.com/products/chinos-blue?variant=47651696836910), click on `size: m`. The page should request `/chinos-blue?section_id=<section_id>__main&option_values=<id_1>,<id_2>,<id_3>`.

**Main product - quick add modal**
- [ ] On [the collections page](https://combined-listings-test.myshopify.com/collections/all), open the quick add modal for `chinos - blue`, then click on `color: orange`. The page should request `/chinos-orange?option_values=<id_1>,<id_2>,<id_3>`
- [ ] On [the collections page](https://combined-listings-test.myshopify.com/collections/all), open the quick add modal for `chinos - blue`, then click on `size: m`. The page should request `/chinos-blue?section_id=<section_id>__main&option_values=<id_1>,<id_2>,<id_3>`.

**Featured product**
_Note -- the option value selection will not work as expected since featured products are not supported in the API. We just want to verify that the request URLs are correct._
- [ ] On the `chinos-black` featured product on [this page](https://combined-listings-test.myshopify.com/pages/contact), select `color: blue`. The page should request `/chinos-blue?section_id=<section_id>__featured_product_<slug>&option_values=<id_1>,<id_2>,<id_3>`
- [ ] Refresh the page and select `size: m` on the `chinos-black` featured product. The page should request `/chinos-black?section_id=<section_id>__featured_product_<slug>&option_values=<id_1>,<id_2>,<id_3>`
- [ ] Further down the page, switch size or color for the `normal product` featured product. The page should request `/normal-product?section_id=<section_id>__featured_product_<slug>&option_values=<id_1>,<id_2>,<id_3>`

### Checklist
- [x] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Review by ludoboludo — COMMENTED** [review]
The feat. product section seem to be missing from the examples you shared. It takes me to a contact page without any feat. product section on it

File: assets/product-info.js [hunk 1]
```diff
@@ -72,21 +72,39 @@ if (!customElements.get('product-info')) {
...
+        this.resetProductFormState();
 
-        const productForm = this.productForm;
-        productForm?.toggleSubmitButton(true);
-        productForm?.handleErrorMessage();
+        // url of the product associated with the selected option value
+        const productUrl = target.dataset.productUrl || this.pendingRequestUrl || this.dataset.url;
+
+        let requestUrl, shouldSwapProduct, shouldFetchFullPage;
+        if (this.isFeaturedProduct) {
```
Comment (ludoboludo) [inline assets/product-info.js:81]: When we're expecting the backend to work for feat. product ?

**Review by ludoboludo — APPROVED** [review]

File: assets/product-info.js (hunk 1 above)
Comment (lhoffbeck) [reply assets/product-info.js:81]: It's kicking off this week

**Review by dan-menard — APPROVED** [review]
Love all the renames 😅

**Pull request merged**